illumio_connector.py

```diff
@@ -374,45 +369,59 @@ def _handle_create_service_binding(self, param):
             param["workload_hrefs"]
         )
+        len_workload_list = len(workload_hrefs_list)
         virtual_service_href = param["virtual_service_href"]
 
         ret_val = self.connect_pce(action_result)
         if phantom.is_fail(ret_val):
             return action_result.get_status()
 
         try:
+            service_bindings = self._pce.service_bindings.create(
+                [
+                    {
+                        "workload": {"href": workload_href},
+                        "virtual_service": {"href": virtual_service_href},
+                        "port_overrides": [],
+                    }
+                    for workload_href in workload_hrefs_list
+                ]
+            )
+
             service_binding_list_for_virtual_service = self._pce.service_bindings.get(
                 params={"virtual_service": virtual_service_href}
             )
             bounded_workload_with_virtual_service_list = [
-                service_binding.to_json()["workload"]["href"]
+                service_binding.workload.href
                 for service_binding in service_binding_list_for_virtual_service
             ]
 
             for workload in bounded_workload_with_virtual_service_list:
                 if workload in workload_hrefs_list:
                     workload_hrefs_list.remove(workload)
 
-            if not workload_hrefs_list:
-                return action_result.set_status(
-                    phantom.APP_ERROR, "Workloads already bounded"
+            if workload_hrefs_list:
+                err_msg = "Unable to bind following workloads: {}".format(
+                    workload_hrefs_list
+                )
+                action_result.set_status(
+                    phantom.APP_SUCCESS,
+                    "Successfully bound {} workload with virtual service. {}".format(
+                        len_workload_list - len(workload_hrefs_list), err_msg
+                    ),
                 )
 
-            service_bindings = self._pce.service_bindings.create(
-                [
-                    {
-                        "workload": {"href": workload_href},
-                        "virtual_service": {"href": virtual_service_href},
-                        "port_overrides": [],
-                    }
-                    for workload_href in workload_hrefs_list
-                ]
-            )
-            self.debug_print(service_bindings)
-            if len(service_bindings["service_bindings"]) == len(workload_hrefs_list):
+            else:
                 action_result.set_status(
                     phantom.APP_SUCCESS,
-                    "Successfully bound workload with virtual service.",
+                    "Successfully bound {} workload with virtual service".format(
+                        len_workload_list
+                    ),
+                )
+
+            if not service_bindings["service_bindings"] and not workload_hrefs_list:
```

Comment:
I don't think we need this block. It's not really an error; we should think of the `service_bindings` POST as idempotent. If all the workloads are already bound, then great.

Reply:
Added successfully bound workloads message in case of already bind workloads and workloads binded at the movement.